File: internal/content/testdata/bench/csharp/nc-construct-tree.cs
public class Solution {
    public TreeNode build_tree(long[] preorder, long[] inorder) {
        var idx = new Dictionary<long, int>();
        for (int i = 0; i < inorder.Length; i++) idx[inorder[i]] = i;
        int pre = 0;
        TreeNode Build(int lo, int hi) {
            if (lo > hi) return null;
            long val = preorder[pre++];
            var node = new TreeNode((int)val);
            int mid = idx[val];
            node.left = Build(lo, mid - 1);
            node.right = Build(mid + 1, hi);
            return node;
        }
        return Build(0, inorder.Length - 1);
    }
}

File: internal/content/testdata/bench/javascript/nc-construct-tree.js
function build_tree(preorder, inorder) {
    const idx = new Map();
    for (let i = 0; i < inorder.length; i++) {
        idx.set(inorder[i], i);
    }
    let pre = 0;
    function build(lo, hi) {
        if (lo > hi) return null;
        const val = preorder[pre++];
        const node = { val: val, left: null, right: null };
        const mid = idx.get(val);
        node.left = build(lo, mid - 1);
        node.right = build(mid + 1, hi);
        return node;
    }
    return build(0, inorder.length - 1);
}
